fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -432,15 +448,31 @@ public Date getLastModifiedDate() {
         }
         LOGGER.debug("Could not get last modified date property for node {}", node);
 
-        final Date createdDate = getCreatedDate();
         if (createdDate != null) {
             LOGGER.trace("Using created date for last modified date for node {}", node);
             return createdDate;
         }
 
         return null;
     }
+    private long getTimestamp(final String property, final long created) throws RepositoryException {
+        LOGGER.trace("Using {} date", property);
+        final long timestamp = getProperty(property).getDate().getTimeInMillis();
+        if (timestamp < created && created > 0L && !isFrozenResource()) {
+            LOGGER.trace("Updating {} with later created date", property);
+            getNode().setProperty(property, created);
+            return created;
+        }
+        return timestamp;
+    }
 
+    /**
+     * Set the last-modified date to the current date.
+     * @throws RepositoryException
+     */
+    public void touch() throws RepositoryException {
```

Comment:
I'm not actually sure I understand this suggestion. The `RepositoryException` is an important part of this method because there may be an access denied exception, which is handled further up the stack.